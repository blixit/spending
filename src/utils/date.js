/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};