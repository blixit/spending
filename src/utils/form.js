/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
