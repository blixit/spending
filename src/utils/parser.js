export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);