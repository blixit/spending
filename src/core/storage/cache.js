import packageJson from 'package.alias.json';
import { parse, stringify } from 'utils/parser';


export const getPrefix = () => {
  // app prefix from env
  const prefix = process.env.REACT_APP_CACHE_PREFIX;

  // app version from package.json
  return prefix + packageJson.version + '_';
};

export const writeCache = (key, value, options) => {
  const cacheKey = getPrefix() + key;
  localStorage.setItem(cacheKey, stringify(value));
};

export const readCache = (key) => {
  const cacheKey = getPrefix() + key;
  return parse(localStorage.getItem(cacheKey));
};
