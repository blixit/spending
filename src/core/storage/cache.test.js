import 'jest-localstorage-mock';

import { parse, stringify } from 'utils/parser';

import { getPrefix, writeCache, readCache } from './cache';

describe('[core/storage/cache] cache', () => {
  describe('getPrefix', () => {
    const prefix = getPrefix();
    expect(prefix).toContain(process.env.REACT_APP_CACHE_PREFIX);
    expect(prefix).toMatchSnapshot();
  });

  describe('write/read data', () => {
    beforeEach(() => {
      localStorage.clear();
    });
  
    it.each([
      'value',
      15,
      { id: 15},
      ['array']
    ])('should writes the local storage properly', (value) => {
      writeCache('test-key', value);
      const expectedValue = parse(localStorage.getItem(getPrefix() + 'test-key'));
      expect(expectedValue).toEqual(value);
    });
  
    it.each([
      'value',
      15,
      { id: 15},
      ['array']
    ])('should reads the local storage properly', (value) => {
      localStorage.setItem(getPrefix() + 'test-key', stringify(value));
      const expectedValue = readCache('test-key');
      expect(expectedValue).toEqual(value);
    });
  });
});
