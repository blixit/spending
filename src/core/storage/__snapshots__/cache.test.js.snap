// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"sp0.1.0_"`;
